### PR TITLE
Add My Queue

### PR DESCRIPTION
**Pull request by conradbeach**
Allows users to add and and remove videos from their queue.

### CODE REVIEW

File: spec/controllers/queue_items_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  describe "POST create" do
+    let(:video) { create(:video) }
+
+    context "when logged in" do
+      let(:user) { create(:user) }
+
+      before do
+        session[:user_id] = user.id
+
+        request.env["HTTP_REFERER"] = video_path(video)
```
Comment (brandoncc) [inline spec/controllers/queue_items_controller_spec.rb]: Why is this line here? Is the referer required? If so, there should be a test covering what happens when it is not present. If not, the line doesn't need to be here. If there is a special reason why it is preferred to have the line here, I would add a comment above it explaining that reason to other developers.

File: spec/controllers/queue_items_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  end
+
+  describe "DELETE destroy" do
+    let(:user) { create(:user) }
+    let(:video_one) { create(:video) }
+    let(:video_two) { create(:video) }
+    let(:video_three) { create(:video) }
+
+    before do
+      @item_one = create(:queue_item, user: user, video: video_one, position: 1)
```
Comment (brandoncc) [inline spec/controllers/queue_items_controller_spec.rb]: You should use `let` instead of instance variables with Rspec.

```ruby
let(:item_one) { create(:queue_item, user: user, video: video_one, position: 1) }
```

File: spec/controllers/queue_items_controller_spec.rb (hunk 1 above)
Comment (conradbeach) [inline spec/controllers/queue_items_controller_spec.rb]: Thanks for pointing that out. No, that line isn't doing anything. I should have removed it during a previous refactor.

File: spec/controllers/queue_items_controller_spec.rb (hunk 2 above)
Comment (conradbeach) [inline spec/controllers/queue_items_controller_spec.rb]: Thanks. Changed.
